doc/Extensions/SNMP-Trap-Handler.md

```diff
@@ -30,23 +50,113 @@ For non-systemd systems, you can edit TRAPDOPTS in the init script in /etc/init.
+
+[Install]
+WantedBy=multi-user.target
+```
+> In Ubuntu 18 is service located by default in ```/etc/systemd/system/multi-user.target.wants/snmptrapd.service```
+
+There is a list of snmptrapd options:
+
+| Option | Description                                                                                      |
+| -------| ------------------------------------------------------------------------------------------------ |
+|   -a   | Ignore authenticationFailure traps. [OPTIONAL]                                                   |
+|   -f   | Do not fork from the shell                                                                       |
+|   -n   | Use numeric addresses instead of attempting hostname lookups (no DNS) [OPTIONAL]                 |
+|   -m   | MIBLIST: use MIBLIST (`FILE1-MIB:FILE2-MIB`). ALL = All MIBS in DIRLIST.                         |
+|   -M   | DIRLIST: use DIRLIST as the list of locations to look for MIBs. Option is not recursive, so you need to specify each DIR individually, separated by `:`. (For example: /opt/librenms/mibs:/opt/librenms/mibs/cisco:/opt/librenms/mibs/edgecos)|                                            
+
+Good practice is to avoid `-m ALL` because then it will try to load all the MIBs in DIRLIST, which
+will typically fail (snmptrapd cannot load that many mibs). Better is to specify the
+exact MIB files defining the traps you are interested in, for example for LinkDown and LinkUp
+traps `-m IF-MIB`. Multiple files can be added, separated with `:`.
+
+If you want to test or store original TRAPS in log then:
+
+Create folder for storing traps for example in file `traps.log`
+
+```
+sudo mkdir /var/log/snmptrap
 
-Reload service files, enable, and start the snmptrapd service:
+```
+
+Add following config to your snmptrapd.service after `ExecStart=/usr/sbin/snmptrapd -f -m ALL -M /opt/librenms/mibs`
+
+```
+-tLf /var/log/snmptrap/traps.log
+
+```
+
+After succesfuly configured service reload service files, enable, and start the snmptrapd service:
 
 ```
 sudo systemctl daemon-reload
 sudo systemctl enable snmptrapd
 sudo systemctl restart snmptrapd
 ```
 
+## Testing 
+
+You can test if your snmptrapd are working correctly very easily, but you need to understand few things.
+
+### Why we need Uptime
+
+When you send a trap, it must of course conform to a set of standards. Every trap needs an uptime value. Uptime is how long the system has been running since boot. Sometimes this is the operating system, other devices might use the SNMP engine uptime. Regardless, a value will be sent.
+
+So what value should you type in the commands below? Oddly enough, simply supplying no value by using two single quotes '' will instruct the command to obtain the value from the operating system you are executing this on.
+
+For those who dig deeper and look at the spooled trap before it's processed will want to understand what type of format it is. Here is an example:
+`DISMAN-EVENT-MIB::sysUpTimeInstance 36:2:40:51.67`
```

Comment:
Worth mentioning that DISMAN-EVENT-MIB::sysUpTimeInstance and SNMPv2-MIB::sysUpTime.0 are the same OID, just depends on the MIBs enabled?

Reply:
Seems to be the same one indeed. Probably depends on the loaded MIBs. But this sentence is not really mandatory to ease the understanding of traps, so I think removing it is even clearer.